infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -533,7 +514,7 @@ private void finishSpanAsCancelled(Span span, String reason) {
   public CompletableFuture<Void> waitRunningAsync(
       List<CompletableFuture<?>> toCancelFutures, KubernetesMachineImpl machine) {
     CompletableFuture<Void> waitFuture =
-        namespace.deployments().waitRunningAsync(machine.getPodName());
+        namespace.deployments().waitRunningAsync(machine.getPodName(), tracerUtil);
```

Comment:
I still think that it is better to let construction phase pass this dependency because passing tracing as `waitRunningAsync` argument violates SRP of this method. This distracts dev from the purpose of this method.

Reply:
Refactored `KubernetesDeployments` to take the utils at creation in commit https://github.com/eclipse/che/pull/12843/commits/8de23f7ae3170c1af09533c0b674ab2af17c8b7a

Reply:
I agree that it's better, especially since tracerUtil is not `nullable` like the normal tracer is.